WordPressAuthenticator/GoogleSignIn/URL+GoogleSignIn.swift

```diff
@@ -0,0 +1,32 @@
+import Foundation
+
+extension URL {
+
+    // TODO: This is incomplete
+    static func googleSignInAuthURL(clientId: String) throws -> URL {
```

Comment:
I used to add all sorts of domain-specific methods to types such as `URL` in the past, then realized it's seldom appropriate because it "leaks" those methods to all the possible consumers of the type. 

For example, in this case, any class that can `import Foundation` and access `URL` will now have the `googleSignInAuthURL(...)` method in its autocompletion list.

In an app, that would definitely be confusing. In a relatively narrow focused library, I think that's acceptable.

However, I'm happy to move this logic in an dedicated object if you think it would be better. Something like `GoogleSignInURLBuilder.makeAuthURL(clientId: ...)`.